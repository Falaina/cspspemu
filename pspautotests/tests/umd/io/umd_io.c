#include <common.h>

#include <pspsdk.h>
#include <pspkernel.h>
#include <pspthreadman.h>
#include <psploadexec.h>
#include <pspumd.h>

int main(int argc, char **argv) {
	int result;
	int directory;
	SceIoDirent directoryEntry;
	
	if ((directory = sceIoDopen("disc0:/PSP_GAME")) > 0) {
		do {
			result = sceIoDread(directory, &directoryEntry);
			printf("%d : '%s'\n", result, directoryEntry.d_name);
		} while(result > 0);
	} else {
		printf("Unknown path\n");
	}

	if ((directory = sceIoDopen("disc0:/unexistant_path")) > 0) {
		do {
			result = sceIoDread(directory, &directoryEntry);
			printf("%d : '%s'\n", result, directoryEntry.d_name);
		} while (result > 0);
	} else {
		printf("Unknown path\n");
	}

	printf("%08X\n", sceIoOpen("_unexistant_file.vag", PSP_O_RDONLY, 0777));
	printf("%08X\n", sceIoOpen("/_unexistant_file.vag", PSP_O_RDONLY, 0777));
	printf("%08X\n", sceIoOpen("disc0:/_unexistant_file.vag", PSP_O_RDONLY, 0777));
	printf("%08X\n", sceIoOpen("disc0:/path/to/unexistant/file.vag", PSP_O_RDONLY, 0777));

	return 0;
}