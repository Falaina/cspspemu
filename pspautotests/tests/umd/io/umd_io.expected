2 : 'PARAM.SFO'
1 : 'SYSDIR'
0 : 'SYSDIR'
Unknown path
80010002
80010002
80010002
80010002